### DISCUSSION TITLE
fan curve not respected

### DISCUSSION THREAD

**Discussion by sn4k3** [seerge/g-helper#2637] [Q&A] [answered]
Model: ASUS A16 FA607PI

50-70ºc = 2600rpms
80ºc = 3000rpms

Actual 61ºc = 3300 rpms (Expected: 2600rpms) This result was taken in idle for sometime. rpms stay fixed and temps from 60ºc to 65ºc.
The value is very off, any idea why?

![image](https://github.com/seerge/g-helper/assets/113281/c0ff1978-ba2b-4a25-bdb0-5609d7298559)

**Comment by seerge** [comment, marked as answer]
@sn4k3 hello, 

RPM values on the Y-axis are indication only as under the hood fan curve is in some abstract % points. Fans are controlled by BIOS so don't expect them to match perfectly. 

If that really bothers you - you can always try [Experimental build with manual fan control](https://github.com/seerge/g-helper/discussions/2272)

Thanks.

**Reply by sn4k3** [reply]
Hello seerge,

Ok I understand now, do you have a clue if the offset is fixed? So, we can use math to discover the actual offset value and tune accordingly? 
Do you have any plans to include your experimental control under current release and probably have a check to switch between control modes?

Regards

**Reply by seerge** [reply]
@sn4k3 it varies a lot from model to model (i.e. from BIOS to BIOS). 

Calibrate button sets fan curve to 100% and measures max RPM trying to match % with RPMs, that's best I can do here.

As for Experimental build - I don't plan to push it live to everyone yet :) as it's more like a last-resort solution for people who are not happy with the way fans are controlled by BIOS

**Reply by sn4k3** [reply]
I have calibrated before because I notice default RPMs was very low, but I'm happy with it, I will just be experimenting.
BTW you think is possible as feture request to have a grid to define values manually? Sometimes I hate graph because mouse, it's very hard to be accurate on points. Something like a right click on graph and display a grid would be awesome.

Also another thing I notice is that when you check/uncheck "Apply custom curve" somehow temp limit is not respected and looks like going to default. Switching profiles also take some time for cut the power, but if I hit apply button it is instant.

Normal behaviour:
![Cinebench_JVMDhp3DuK](https://github.com/seerge/g-helper/assets/113281/f8b3ed42-6a2a-4eeb-8432-1a7949c1a333)

After clicking the checkbox of fan curve (temp limit is lost):
![GHelper_9oUIRsNjhl](https://github.com/seerge/g-helper/assets/113281/5433883a-2fd8-4f52-9fc8-28692e8d2e59)

After click "apply", temp limit is now respected and stay like that as long I dont mess with the fan curve checkbox
![GHelper_TC2RF5AFOR](https://github.com/seerge/g-helper/assets/113281/4964a9b4-074c-4a2b-8692-955aeaedb371)

**Reply by seerge** [reply]
> Sometimes I hate graph because mouse, it's very hard to be accurate on points. Something like a right click on graph and display a grid would be awesome.

There is not much sense, cause BIOS may simply override fans when it feels to. In AC you had some abstract % on the chart at all that were not reflecting actual speed.

**Reply by sn4k3** [reply]
Yes, I mean, having the graph layout in addition to a grid layout, something like this, but with `NumericInputBox` :

![image](https://github.com/seerge/g-helper/assets/113281/276612de-e0c5-4901-a1b2-88d31eb1fa55)

Eg: Right click on graph to change display modes. Graphs for me was always hard to deal with fine precision. Also much more fast for fast tune in my case.
